src/App.jsx

```diff
@@ -1,34 +1,52 @@
 import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+
 
 const App = () => {
   const [users, setUsers] = useState([]);
+  const [usersSearch,setUsersSearch]=useState([])
 
   useEffect(() => {
     fetchUsers();
   }, []);
 
   const fetchUsers = async () => {
     try {
-      const response = await get('https://jsonplaceholder.typicode.com/users');
-      if (!response.ok) {
+      const response = await axios.get('https://jsonplaceholder.typicode.com/users');
+      if (response.status!=200) {
         throw new Error('Failed to fetch users');
       }
-      const data = await response.json();
+      const data = await response.data;
       setUsers(data);
     } catch (error) {
       console.error(error);
     }
   };
-
+  
+  const onChange=( event)=>{
+      const usersUpdate = users.filter((item)=>item.name.includes(event.target.value))
+      setUsersSearch(usersUpdate)
+  }
+  
   return (
     <div>
       <h1>Lista de Usuarios</h1>
       <input
+        name='nameUser'
         type="text"
         placeholder="Buscar por nombre o correo electrónico"
+        onChange={onChange}
       />
-      <ul>
-        {users.map((user) => (
+      <ul>        
+        {usersSearch.length!=0?
```

Comment:
aqui puedes hacer una refactorizacion, hacer un users.filter(.....).map(()=> {<div>.....}). entonces para este caso no tienes que hacerle un operador ternario. y recuerda que cada vez que se llama un setState se renderiza tdo el componente. 





Mira la solucion:














Primero intentalo tu y despues ves esto,


`import React, { useState, useEffect } from 'react';

const App = () => {
  const [users, setUsers] = useState([]);
  const [searchTerm, setSearchTerm] = useState('');

  useEffect(() => {
    fetchUsers();
  }, []);

  const fetchUsers = async () => {
    try {
      const response = await fetch('https://jsonplaceholder.typicode.com/users');
      if (!response.ok) {
        throw new Error('Failed to fetch users');
      }
      const data = await response.json();
      setUsers(data);
    } catch (error) {
      console.error(error);
    }
  };

  const handleSearch = (event) => {
    setSearchTerm(event.target.value);
  };

  const filteredUsers = users.filter((user) => {
    const { name, email } = user;
    return (
      name.toLowerCase().includes(searchTerm.toLowerCase()) ||
      email.toLowerCase().includes(searchTerm.toLowerCase())
    );
  });

  return (
    <div>
      <h1>Lista de Usuarios</h1>
      <input
        type="text"
        placeholder="Buscar por nombre o correo electrónico"
        value={searchTerm}
        onChange={handleSearch}
      />
      <ul>
        {filteredUsers.map((user) => (
          <li key={user.id}>
            <strong>{user.name}</strong>
            <p>{user.email}</p>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default App;`